### PR TITLE
Feature/molvis refactoring

### PR DESCRIPTION
**Pull request by martinfalk**
* refactored molecular rendering into two options
  1. direct rendering of a `MolecularStructure` using a `Molecular Renderer` processor
  2. indirect rendering of a molecular mesh with `Molecular Mesh Renderer`
* added picking/tooltips for atoms
* utility functions for amino acids and chains
* different commonly-used colormaps (atoms & residues)
* regression tests and example pdb files

### CODE REVIEW

File: molvis/molvisbase/src/io/basicpdbreader.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                parseException(tag, "invalid position");
+            }
+            double [[maybe_unused]] occupancy = 0.0;
+            if (!detail::fromStr(line.substr(54, 6), occupancy)) {
+                parseException(tag, "invalid occupancy");
+            }
+            double bFactor = 0.0;
+            if (!detail::fromStr(line.substr(60, 6), bFactor)) {
+                parseException(tag, "invalid temperature factor");
+            }
```
Comment (petersteneteg) [inline molvis/molvisbase/src/io/basicpdbreader.cpp:149]: could extract this into function
```
template <typename T>
T parseSection(string_view line, size_t begin, size_t size, sting_view tag, string_view desc) {
    T res;
    if (!detail::fromStr(line.substr(start, size), res)) {
        parseException(tag, desc);
    }
    return res;
}
```
then you could just write 
```
const auto occupancy = parseSection<double>(line, 54, 6, tag, "invalid occupancy");
const auto Bfactor = parseSection<double>(line, 60, 6, tag, "invalid temperature factor");
```
etc...

File: molvis/molvisbase/src/io/basicpdbreader.cpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                data.residues.push_back({residueId, molvis::aminoacid::fromFullName(residueName),
+                                         std::string(residueName), chainId});
+            }
+            data.atoms.positions.push_back(pos);
+            data.atoms.bFactors.push_back(bFactor);
+            data.atoms.modelIds.push_back(currentModel);
+            data.atoms.chainIds.push_back(chainId);
+            data.atoms.residueIds.push_back(residueId);
+            data.atoms.atomicNumbers.push_back(molvis::element::fromAbbr(element));
+            data.atoms.fullNames.push_back(std::string(fullName));
```
Comment (petersteneteg) [inline molvis/molvisbase/src/io/basicpdbreader.cpp:168]: emplace_back, then you don't need the std::string

File: molvis/molvisgl/src/processors/molecularrenderer.cpp [hunk 3]
```diff
@@ -124,58 +162,55 @@ void MolecularRenderer::process() {
...
-            throw Exception("Unsupported representation: 'Ribbon'", IVW_CONTEXT);
-            break;
-        case Representation::Cartoon:
-            throw Exception("Unsupported representation: 'Cartoon'", IVW_CONTEXT);
-            break;
-        default:
-            drawVdW(radiusScaling_);
-            break;
+    for (auto structure : inport_) {
+        auto mesh = createMesh(*structure);
```
Comment (petersteneteg) [inline molvis/molvisgl/src/processors/molecularrenderer.cpp:190]: is this something you want to do every frame? seems expensive

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: molvis/molvisbase/include/inviwo/molvisbase/util/aminoacid.h [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+namespace molvis {
+
+// clang-format off
+enum class AminoAcid : unsigned char {
+    Unknown = 0,
+    Ala, Arg, Asn, Asp, Cys, Gln, Glu, Gly,
+    His, Ile, Leu, Lys, Met, Phe, Pro, Ser,
+    Thr, Trp, Tyr, Val, Sec, Pyl, Asx, Glx,
+    Xle,
```
[the remark this answers is not in the document]
Comment (martinfalk) [reply molvis/molvisbase/include/inviwo/molvisbase/util/aminoacid.h:53]: that is allowed according to the standard :)

File: molvis/molvisbase/src/io/basicpdbreader.cpp [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+#include <sstream>
+#include <charconv>
+#include <fmt/format.h>
+
+#pragma optimize("", off)
+
+namespace inviwo {
+
+BasicPDBReader::BasicPDBReader() {
+    addExtension(FileExtension("pdb", "Protein Databank files (pdb)"));
```
[the remark this answers is not in the document]
Comment (martinfalk) [reply molvis/molvisbase/src/io/basicpdbreader.cpp:51]: no, that should be fine I think. Needs testing though.

File: molvis/molvisbase/src/util/utilities.cpp [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+#include <inviwo/molvisbase/util/utilities.h>
+
+namespace inviwo {
+
+namespace molvisutil {
+
+namespace detail {
+
+constexpr vec3 hsvToRgb(vec3 c) noexcept {
```
[the remark this answers is not in the document]
Comment (martinfalk) [reply molvis/molvisbase/src/util/utilities.cpp:38]: removed.

File: molvis/molvisbase/src/util/utilities.cpp [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    vec4 K(1.0f, 2.0f / 3.0f, 1.0f / 3.0f, 3.0f);
+    vec3 p = abs(fract(vec3(c.x) + vec3(K)) * 6.0f - vec3(K.w));
+    return c.z * mix(vec3(K.x), clamp(p - vec3(K.x), 0.0f, 1.0f), c.y);
+}
+
+constexpr vec3 hclToRgb(vec3 hcl) noexcept {
+    constexpr float hclgamma = 3;
+    constexpr float hcly0 = 100;
+    constexpr float hclmaxL = 0.530454533953517f;  // == exp(hclgamma / hcly0) - 0.5
```
[the remark this answers is not in the document]
Comment (martinfalk) [reply molvis/molvisbase/src/util/utilities.cpp:47]: `exp` not being constexpr

**Review by petersteneteg — APPROVED** [review]

**Review by petersteneteg — APPROVED** [review]

**Pull request merged**